Extreme busy wait

I was trying out the Timer class, running an exact copy of the version in the doc: http://www.ponylang.org/ponyc/time-Timer/#timer

```pony
use "time"

actor Main
  new create(env: Env) =>
    let timers = Timers
    let timer = Timer(Notify(env), 5_000_000_000, 2_000_000_000)
    timers(consume timer)

class Notify is TimerNotify
  let _env: Env
  var _counter: U32 = 0
  new iso create(env: Env) =>
    _env = env
  fun ref apply(timer: Timer, count: U64): Bool =>
    _env.out.print(_counter.string())
    _counter = _counter + 1
    true
```
I expected cmd to output numbers every few seconds with little to no CPU utilization.

Cmd did display the numbers, but at 50% CPU utilization. When I ran the program from Git Bash (MinGW64) there was no output. I suspect the overhead of printing to the MinGW console was so high the busy waiting threads were completely starving out the writer.

I'm using Windows 7 64 bit, Intel i7-2630QM 2 GHz 4 core 8 thread CPU, 8 GB RAM.
$ ponyc -v
0.10.0-1c33065 [release]
compiled with: llvm 3.9.0 -- ?

Thanks for sending me that output.  The crucial error is:

    LINK : warning LNK4044: unrecognized option '/WHOLEARCHIVE:libponyrt.lib'; ignored

It looks like you have a version of Visual Studio 2015 prior to Update 2, since the `/WHOLEARCHIVE` flag was only introduced to `link.exe` in Update 2.  Can you update to the latest VS2015 and try again, please?

--- a/src/libponyrt/sched/cpu.c
+++ b/src/libponyrt/sched/cpu.c
@@ -305,7 +305,7 @@ void ponyint_cpu_core_pause(uint64_t tsc, uint64_t tsc2, bool yield)
   DTRACE1(CPU_NANOSLEEP, ts.tv_nsec);
   nanosleep(&ts, NULL);
 #else
-  Sleep(0);
+  Sleep(yield ? 1 : 0);
 #endif
 }
 

